### DISCUSSION TITLE
How can I change the request header content-type to application/vnd.api+json to reflect in Open API

### DISCUSSION THREAD

**Discussion by saravanaemis** [fastapi/fastapi#6683] [Questions]
Example:

```
from fastapi import FastAPI
from fastapi.responses import Response, JSONResponse
from pydantic import BaseModel, create_model

app = FastAPI()

class Child(BaseModel):
    title: str
    name: str


class BaseParent1(BaseModel):
    data: create_model('BaseParent', type=(str, ...), attributes=(Child, ...))

@app.post("/items/")
async def create_item(item: BaseParent1):
    return item
```

And my openAPI screenshot as below,
<img width="1380" alt="Screenshot 2021-06-15 at 7 06 39 PM" src="https://user-images.githubusercontent.com/61677324/122062553-24aac480-ce0d-11eb-8394-951897911f9c.png">

Description
If you look into the above openAPI screenshot, By default content type set to 'application/json', 
But I want my request content type should be as 'application/vnd.api+json'. How do I change it to 'application/vnd.api+json'?

**Comment by fonzcastellanos** [comment]
I believe you need to declare the `item` parameter as a `Body` and set the media type.
```python
from fastapi import Body

...

@app.post("/items/")
async def create_item(item: BaseParent1 = Body(..., media_type="application/vnd.api+json", embed=True)):
    return item
```

**Comment by saravanaemis** [comment]
@fonzcastellanos Great 😊😊😊, It's worked I can able to see "application/vnd.api+json" in OpenAPI. But I'm getting below while posting my resquest, if i pass 'Body(..., media_type="application/vnd.api+json", embed=True)' to BaseParent1.
```
{
    "detail": [
        {
            "loc": [
                "body",
                "item"
            ],
            "msg": "field required",
            "type": "value_error.missing"
        }
    ]
}
```
And this is my request payload,
```
{
    "data": {
        "type": "resourceErn",
        "attributes": {
            "title": "test002",
            "name": "test"
        }
    }
}
```

Do I have to add anything to the Body() instance?

**Comment by frankie567** [comment]
With the `embed=True` argument, FastAPI expects you wrap the object in an "item" property (see https://fastapi.tiangolo.com/tutorial/body-multiple-params/#embed-a-single-body-parameter):

```json
{
    "item": {
        "data": {
            "type": "resourceErn",
            "attributes": {
                "title": "test002",
                "name": "test"
            }
        }
    }
}
```

If it's not what you want, just remove the `embed` argument and it should work.
